app/javascript/components/ui/Table.tsx

```diff
@@ -0,0 +1,168 @@
+  const { headerLabels: headers, setHeaderLabels: setHeaders } = useTable();
+
+  React.useEffect(() => {
+    const isHeaderRow = React.Children.toArray(children).every(
+      (child) => React.isValidElement(child) && child.type === TableHead,
+    );
+    if (!isHeaderRow) return;
+
+    setHeaders(
+      React.Children.toArray(children).map<string | null>((child) => {
+        if (React.isValidElement(child) && child.type === TableHead) {
+          // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+          const props = child.props as React.ComponentProps<typeof TableHead>;
+          return typeof props.children === "string" ? props.children : null;
+        }
+        return null;
+      }),
+    );
+  }, [children]);
+
+  const childrenWithLabels = React.Children.map(children, (child, index) => {
+    if (React.isValidElement(child) && child.type === TableCell) {
+      // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      const cell = child as React.FunctionComponentElement<React.ComponentProps<typeof TableCell>>;
+      const label = cell.props.label ?? headers?.[index] ?? null;
+      return React.cloneElement(cell, label ? { label } : {});
+    }
+    return child;
+  });
+
+  return (
+    <tr
+      aria-selected={selected}
+      className={classNames(
+        "block rounded-sm border border-border lg:table-row [tbody_>_&]:bg-background",
+        selected && "cursor-pointer bg-active-bg",
+        className,
+      )}
+      {...props}
+    >
+      {childrenWithLabels}
+    </tr>
+  );
+};
+
+const cellRoundingClasses =
+  "lg:[table_>_:last-child_>_tr:last-child_>_&:first-child]:rounded-bl-sm lg:[table_>_:last-child_>_tr:last-child_>_&:last-child]:rounded-br-sm";
+
+export const TableHead = ({
+  className,
+  scope,
+  sortDirection,
+  onSort,
+  children,
+  ...props
+}: Omit<React.ThHTMLAttributes<HTMLTableCellElement>, "aria-sort"> & {
+  sortDirection?: "ascending" | "descending" | "none";
+  onSort?: () => void;
+}) => (
+  <th
+    aria-sort={sortDirection}
+    scope={scope}
+    onClick={onSort}
+    className={classNames(
+      "px-4 py-3 text-left align-middle lg:table-cell lg:whitespace-nowrap lg:[tbody_>_tr_>_&]:border-t lg:[tbody_>_tr_>_&]:border-border",
+      cellRoundingClasses,
+      scope === "row" && "font-normal",
+      sortDirection && "cursor-pointer",
+      className,
+    )}
+    {...props}
+  >
+    <span className="inline-flex items-center gap-1">
+      {children}
+      {sortDirection && sortDirection !== "none" ? (
+        <span className="inline-block">{sortDirection === "ascending" ? "↑" : "↓"}</span>
+      ) : null}
+    </span>
+  </th>
+);
+
+export const TableCell = ({
+  className,
+  actions,
+  hideLabel,
+  label,
+  children,
+  ...props
+}: Omit<React.TdHTMLAttributes<HTMLTableCellElement>, "aria-busy"> & {
+  actions?: boolean;
```

Comment:
Would probably use `variant="actions"` for this (`actions` reads like it would take a `ReactNode`)